backend/internal/service/oidc_service.go

```diff
@@ -277,6 +278,93 @@ func (s *OidcService) createTokenFromRefreshToken(refreshToken, clientID, client
 	return accessToken, newRefreshToken, 3600, nil
 }
 
+func (s *OidcService) IntrospectToken(clientID, clientSecret, tokenString string) (introspectDto dto.OidcIntrospectionResponseDto, err error) {
+	if clientID == "" || clientSecret == "" {
+		return introspectDto, &common.OidcMissingClientCredentialsError{}
+	}
+
+	// Get the client to check if we are authorized.
+	var client model.OidcClient
+	if err := s.db.First(&client, "id = ?", clientID).Error; err != nil {
+		return introspectDto, &common.OidcClientSecretInvalidError{}
+	}
+
+	// Verify the client secret. This endpoint may not be used by public clients.
+	if client.IsPublic {
+		return introspectDto, &common.OidcClientSecretInvalidError{}
+	}
+
+	if err := bcrypt.CompareHashAndPassword([]byte(client.Secret), []byte(clientSecret)); err != nil {
+		return introspectDto, &common.OidcClientSecretInvalidError{}
+	}
+
+	token, err := s.jwtService.VerifyIdToken(tokenString, false)
```

Comment:
```suggestion
	token, err := s.jwtService.VerifyOauthAccessToken(tokenString, false)
```

Shouldn't we verify if it is an access token instead?

Reply:
@stonith404 I adjusted it in commit c925831e. It's a bit confusing that there is `VerifyAccessToken`, `VerifyIdToken` and `VerifyOauthAccessToken`. Looks like `VerifyAccessToken` is strictly meant for access towards PocketId itself. The different between `VerifyIdToken` and `VerifyOauthAccessToken` seems to be mainly that the former allows for less strict validation. But at least semantically `VerifyOauthAccessToken` makes indeed much more sense.

Reply:
@aksdb Thanks. Yes, that's a bit concerning because at first glance, it seems possible that an `id_token` could be successfully validated by `VerifyOauthAccessToken`, which would be very bad. I'll look into it as soon as possible. After that, the PR should be ready to merge.